core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/validation/ContractValidationServiceImpl.java

```diff
@@ -131,7 +132,7 @@ public boolean validate(ClaimToken token, ContractAgreement agreement, ContractO
     @NotNull
     private ArrayList<Criterion> createCriteria(ContractOffer offer, ContractDefinition contractDefinition) {
         var criteria = new ArrayList<>(contractDefinition.getSelectorExpression().getCriteria());
-        var criterion = new Criterion(Asset.PROPERTY_ID, "=", offer.getAsset().getId());
+        var criterion = new Criterion(Asset.PROPERTY_ID, "=", "'" + offer.getAsset().getId() + "'");
```

Comment:
I assume this would break several existing implementations that expect the asset id in the right operand. Is this a workaround that is necessary to send the asset id somewhere else? If yes maybe there is a better place for this workaround.

Reply:
I agree. As far as I recall this has to do with Cosmos (or SQL in general) expecting string comparisons to be put in quotes. It might prove difficult to decide when to add the ticks and when not, because in order to do that reliably we'd need schema introspection.